module-2/cdk/lib/ECSConstruct.ts

```diff
@@ -0,0 +1,179 @@
+    const EcsServiceRole = new iam.Role(this, "EcsServiceRole", {
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal("ecs.amazonaws.com"),
+        new iam.ServicePrincipal("ecs-tasks.amazonaws.com")
+      ),
+      path: "/"
+    });
+    const EcsServiceRoleStatement1 = new iam.PolicyStatement();
+    EcsServiceRoleStatement1.addActions(
+      // Rules which allow ECS to attach network interfaces to instances
+      // on your behalf in order for awsvpc networking mode to work right
+      "ec2:AttachNetworkInterface",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateNetworkInterfacePermission",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteNetworkInterfacePermission",
+      "ec2:Describe*",
+      "ec2:DetachNetworkInterface",
+
+      // Rules which allow ECS to update load balancers on your behalf
+      //  with the information sabout how to send traffic to your containers
+      "elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
+      "elasticloadbalancing:DeregisterTargets",
+      "elasticloadbalancing:Describe*",
+      "elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+      "elasticloadbalancing:RegisterTargets",
+
+      //  Rules which allow ECS to run tasks that have IAM roles assigned to them.
+      "iam:PassRole",
+
+      //  Rules that let ECS interact with container images.
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+
+      //  Rules that let ECS create and push logs to CloudWatch.
+      "logs:DescribeLogStreams",
+      "logs:CreateLogStream",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents"
+    );
+    EcsServiceRoleStatement1.addAllResources();
+    EcsServiceRole.addToPolicy(EcsServiceRoleStatement1);
+
+    this.cluster = new ecs.Cluster(this, "Cluster", {
+      clusterName: "MythicalMysfits-Cluster",
+      vpc: core.vpc
+    });
+    this.cluster.connections.allowFromAnyIPv4(new ec2.TcpPort(8080));
+
+    // Instantiate Amazon ECS Service with an automatic load balancer
+    this.ecsService = new LoadBalancedFargateService(this, "Service", {
+      cluster: this.cluster,
+      loadBalancerType: ecsPatterns.LoadBalancerType.Network,
+      desiredCount: 1,
+      createLogs: true,
+      publicLoadBalancer: true,
+      containerPort: 8080,
+      image: ecs.ContainerImage.fromEcrRepository(ecr.repository),
+      executionRole: EcsServiceRole,
+      taskRole: ECSTaskRole
+    });
+    this.ecsService.service.connections.allowFrom(
+      new ec2.CidrIPv4(core.vpc.vpcCidrBlock),
+      new ec2.TcpPort(8080)
+    );
+  }
+}
+interface MythicalLoadBalancedFargateServiceProps extends ecsPatterns.LoadBalancedFargateServiceProps {
+  executionRole?: Role;
+  taskRole?: Role;
+}
+
+/**
+ * A Fargate service running on an ECS cluster fronted by a load balancer
+ */
+export class LoadBalancedFargateService extends ecsPatterns.LoadBalancedServiceBase {
```

Comment:
Why do we need to extend this class and not just compose it into the class above? Seems like a redundant layer.